packages/components/Callout/macos/CalloutWindow.swift

```diff
@@ -22,11 +22,36 @@ class CalloutWindow: NSWindow {
 		// Dismiss the Callout if the window is no longer active.
 		NotificationCenter.default.addObserver(self, selector: #selector(dismissCallout), name: NSApplication.didResignActiveNotification, object: nil)
 
-		// Dismiss the Callout if the user touched/clicked outside the bounds of the callout.
+		// Dismiss the Callout if the user touched/clicked outside the Callout Window or any of our child windows.
 		mouseEventMonitor = NSEvent.addLocalMonitorForEvents(matching: .leftMouseUp, handler: { [weak self] (event) -> NSEvent? in
-			if (event.window != self) {
+			guard let clickedWindow = event.window else {
+				return event
+			}
+
+			var shouldDismissCallout = false
+
+			if (clickedWindow.isEqual(to: self) ) {
+				shouldDismissCallout = false
+			} else if (clickedWindow.isMainWindow) {
+				shouldDismissCallout = true
+			} else if (clickedWindow.isEqual(to: self?.parent) && self?.parent is CalloutWindow) {
+				// We are a child window of a Callout (e.g: a ContextualMenu Submenu), and the click happened in our parent
+				shouldDismissCallout = false
+			} else {
+				// Check if the click happened in any of our child windows (e.g: our submenus)
```

Comment:
I feel a little iffy mentioning Menus/submenus inside Callout (since Callout doesn't know / shouldn't know about what it's hosting), but this was the best way I could describe these cases.